cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -944,7 +944,7 @@ void RegisterScalarStringAscii(FunctionRegistry* registry) {
   MakeUnaryStringBatchKernel<AsciiUpper>("ascii_upper", registry);
   MakeUnaryStringBatchKernel<AsciiLower>("ascii_lower", registry);
 
-  AddUnaryStringPredicate<IsAscii>("binary_isascii", registry);
+  AddUnaryStringPredicate<IsAscii>("string_isascii", registry);
```

Comment:
Why isn't this `string_is_ascii`? It seems we prefer `is_x` to `isx` (cf. `isin` -> `is_in` in this PR).

Reply:
That's a good point. Should we make a followup PR?
(if it's me, it's only tomorrow)

Reply:
I can make this change today unless someone else wants to

Reply:
Also `ascii_is_alnum`, etc. ;-)